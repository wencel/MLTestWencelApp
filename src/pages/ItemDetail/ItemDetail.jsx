import PropTypes from 'prop-types';
import SearchNavbar from 'components/SearchNavbar';
import { useEffect } from 'react';
import ItemDetailCard from 'components/ItemDetailCard';
import BreadCrums from 'components/BreadCrums';
import ContentContainer from 'components/ContentContainer';

const ItemDetail = ({ item, redirectSearchItems, requestFetchItem, id }) => {
  useEffect(() => {
    if (id) requestFetchItem(id);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [id]);
  return (
    <>
      <SearchNavbar onSearch={redirectSearchItems} />
      <BreadCrums categories={[]} />
      <ContentContainer>
        <ItemDetailCard
          item={item.data}
          loading={item.loading}
          error={item.error}
        />
      </ContentContainer>
    </>
  );
};

ItemDetail.propTypes = {
  id: PropTypes.string,
  item: PropTypes.object,
  redirectSearchItems: PropTypes.func,
  requestFetchItem: PropTypes.func,
};

ItemDetail.defaultProps = {
  redirectSearchItems: () => {},
  requestFetchItem: () => {},
};

export default ItemDetail;
