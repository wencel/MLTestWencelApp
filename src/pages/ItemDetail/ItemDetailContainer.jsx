import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import {
  redirectSearchItems,
  requestFetchItem,
} from 'reducers/item/itemActions';
import { selectedItemSelector } from 'reducers/item/itemSelectors';
import ItemDetail from './ItemDetail';

const ItemDetailContainer = ({
  redirectSearchItems,
  requestFetchItem,
  item,
}) => {
  return (
    <ItemDetail
      redirectSearchItems={redirectSearchItems}
      requestFetchItem={requestFetchItem}
      item={item}
    />
  );
};

ItemDetailContainer.propTypes = {
  item: PropTypes.object,
  redirectSearchItems: PropTypes.func,
  requestFetchItem: PropTypes.func,
};

const mapStateToProps = state => {
  return {
    item: selectedItemSelector(state),
  };
};

const mapDispatchToProps = dispatch => ({
  redirectSearchItems: data => {
    dispatch(redirectSearchItems(data));
  },
  requestFetchItem: data => {
    dispatch(requestFetchItem(data));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ItemDetailContainer);
