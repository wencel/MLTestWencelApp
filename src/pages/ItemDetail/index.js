import ItemDetail from './ItemDetailContainer';
export default ItemDetail;
