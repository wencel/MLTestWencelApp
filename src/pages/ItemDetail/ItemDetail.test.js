import React from 'react';
import { mount } from 'enzyme';
import ItemDetail from './ItemDetail';

describe('WHEN mounting the component', () => {
  const redirectSearchItems = jest.fn(0);
  const requestFetchItem = jest.fn(0);
  const item = {
    data: {
      id: 'MLA906703376',
      title: 'Nintendo Switch 32gb Standard Rojo Neón, Azul Neón Y Negro',
      price: {
        currency: 'ARS',
        amount: 59300,
        decimals: '00',
      },
      picture: 'http://http2.mlstatic.com/D_883371-MLA32731749246_112019-I.jpg',
      condition: 'new',
      free_shipping: true,
      state: 'Buenos Aires',
    },
    loading: false,
    error: '',
  };
  const wrapper = mount(
    <ItemDetail
      id='testID'
      redirectSearchItems={redirectSearchItems}
      requestFetchItem={requestFetchItem}
      item={item}
    />
  );
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('calls the requestFetchItem function when on mount', () => {
    // Testing the useEffect requires a timeout for it to work
    setTimeout(() => {
      expect(requestFetchItem).toHaveBeenCalledTimes(1);
    }, 0);
  });
});
