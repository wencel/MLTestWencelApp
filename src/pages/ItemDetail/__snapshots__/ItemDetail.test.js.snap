// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN mounting the component renders correctly 1`] = `ReactWrapper {}`;
